### PR TITLE
Document more exceptions

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/service/src/Exceptions/IotHubServiceErrorCode.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ namespace Microsoft.Azure.Devices
     public enum IotHubServiceErrorCode
     {
         /// <summary>
-        /// Used when the error code returned by the hub is unrecognized. If encountered, please report the issue so it can be added here.
+        /// Used when the error code returned by IoT hub is not recognized. See the error message for more details.
```
Comment (patilsnr) [inline iothub/service/src/Exceptions/IotHubServiceErrorCode.cs:12]: I forget, do we capitalize `hub`?

**Review by patilsnr — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: SDK v2 migration guide.md [hunk 2]
```diff
@@ -197,8 +197,10 @@ Find a client you currently use below, read the table of API name changes and us
 | `MethodRequest` | `DirectMethodRequest` | Use full name of the operation type.⁴ |
 | `MethodResponse` | `DirectMethodResponse` | See⁴ |
 | `IotHubException` | `IotHubClientException` | Specify the exception is for Hub device and module client only. |
-| `AuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming.⁵ |
-| `ClientAuthenticationWithToken` | `ClientAuthenticationWithSharedAccessSignature` | See⁵ |
+| `DeviceAuthenticationWithTokenRefresh.` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming and reduce duplication.⁵ |
```
Comment (brycewang-microsoft) [inline SDK v2 migration guide.md:200]:
```suggestion
| `DeviceAuthenticationWithTokenRefresh` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming and reduce duplication.⁵ |
```

File: SDK v2 migration guide.md [hunk 3]
```diff
@@ -197,8 +197,10 @@ Find a client you currently use below, read the table of API name changes and us
...
+| `DeviceAuthenticationWithToken` and `ModuleAuthenticationWithToken` | `ClientAuthenticationWithSharedAccessSignature` | See⁵ |
+| `DeviceAuthenticationWithSakRefresh` and `ModuleAuthenticationWithSakRefresh` | `ClientAuthenticationWithTokenRefresh` | See⁵ |
```
Comment (brycewang-microsoft) [inline SDK v2 migration guide.md:202]: I doubt this, should we exchange the newer API names between L200 and L202?

File: SDK v2 migration guide.md [hunk 4]
```diff
@@ -197,8 +197,10 @@ Find a client you currently use below, read the table of API name changes and us
...
+| `DeviceAuthenticationWithTokenRefresh.SafeCreateNewToken(...)` and `ModuleAuthenticationWithTokenRefresh.SafeCreateNewToken(...)` | `ClientAuthenticationWithSharedAccessKeyRefresh.SafeCreateNewTokenAsync(...)` | Async suffix for async methods. |
```
Comment (brycewang-microsoft) [inline SDK v2 migration guide.md:203]: Is it supposed to be like this?

```suggestion
| `DeviceAuthenticationWithSakRefresh.SafeCreateNewToken(...)` and `ModuleAuthenticationWithSakRefresh.SafeCreateNewToken(...)` | `ClientAuthenticationWithSharedAccessKeyRefresh.SafeCreateNewTokenAsync(...)` | Async suffix for async methods. |
```

File: SDK v2 migration guide.md (hunk 3 above)
Comment (drwill-ms) [reply SDK v2 migration guide.md:202]: You're right. Fixed.

File: SDK v2 migration guide.md (hunk 4 above)
Comment (drwill-ms) [reply SDK v2 migration guide.md:203]: Sort of. Fixed.

File: SDK v2 migration guide.md [hunk 5]
```diff
@@ -197,8 +197,10 @@ Find a client you currently use below, read the table of API name changes and us
 | `MethodRequest` | `DirectMethodRequest` | Use full name of the operation type.⁴ |
 | `MethodResponse` | `DirectMethodResponse` | See⁴ |
 | `IotHubException` | `IotHubClientException` | Specify the exception is for Hub device and module client only. |
-| `AuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming.⁵ |
-| `ClientAuthenticationWithToken` | `ClientAuthenticationWithSharedAccessSignature` | See⁵ |
+| `DeviceAuthenticationWithTokenRefresh` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming and reduce duplication.⁵ |
```
Comment (brycewang-microsoft) [inline SDK v2 migration guide.md:200]:
```suggestion
| `DeviceAuthenticationWithTokenRefresh` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithTokenRefresh` | More descriptive naming and reduce duplication.⁵ |
```

Comment (drwill-ms) [reply SDK v2 migration guide.md:200]: Sloppy, sorry.

**Pull request merged**